src/framework/theme/components/window/window.service.ts

```diff
@@ -0,0 +1,166 @@
+} from '@angular/core';
+import { filter } from 'rxjs/operators';
+import {
+  NbBlockScrollStrategy, NbComponentPortal, NbComponentType, NbOverlayPositionBuilder, NbOverlayRef, NbOverlayService,
+} from '../cdk/overlay';
+import {
+  NB_DEFAULT_WINDOWS_CONFIG, NB_WINDOW_CONTENT, NbWindowConfig, NbWindowState, NB_WINDOW_CONTEXT,
+} from './window-types';
+import { NbWindowRef } from './window-ref';
+import { NbWindowsContainerComponent } from './windows-container.component';
+import { NbWindowComponent } from './window.component';
+
+/**
+ * The `NbWindowService` can be used to open windows.
+ *
+ * @stacked-example(Showcase, window/window-showcase.component)
+ *
+ * A new window can be opened by calling the `open` method with a component or template to be loaded
+ * and an optional configuration.
+ * `open` method will return `NbWindowRef` that can be used for the further manipulations.
+ *
+ * ```ts
+ * const windowRef = this.windowService.open(MyComponent, { ... });
+ * ```
+ *
+ * `NbWindowRef` gives you ability manipulate opened window.
+ * Also, you can inject `NbWindowRef` inside provided component which rendered in window.
+ *
+ * ```ts
+ * this.windowService.open(MyWindowComponent, { ... });
+ *
+ * // my.component.ts
+ * constructor(protected windowRef: NbWindowRef) {
+ * }
+ *
+ * minimize() {
+ *   this.windowRef.minimize();
+ * }
+ *
+ * close() {
+ *   this.windowRef.close();
+ * }
+ * ```
+ *
+ * Instead of component you can create window from TemplateRef. As usual you can access context provided via config
+ * via `let-` variables. Also you can get reference to the `NbWindowRef` in context's `windowRef` property.
+ *
+ * @stacked-example(Template ref, window/template-window.component)
+ *
+ * ### Configuration
+ *
+ * As mentioned above, `open` method of the `NbWindowService` may receive optional configuration options.
+ * Also, you can modify default windows configuration through `NbWindowModule.forRoot({ ... })`.
+ * You can read about all available options on [API tab](docs/components/window/api#nbwindowconfig).
+ *
+ * @stacked-example(Configuration, window/windows-backdrop.component)
+ */
+@Injectable()
+export class NbWindowService {
+
+  protected overlayRef: NbOverlayRef;
+  protected windowsContainerViewRef: ViewContainerRef;
+  protected openWindows: NbWindowRef[] = [];
+
+  constructor(
+    protected componentFactoryResolver: ComponentFactoryResolver,
+    protected overlayService: NbOverlayService,
+    protected overlayPositionBuilder: NbOverlayPositionBuilder,
+    protected blockScrollStrategy: NbBlockScrollStrategy,
+    @Inject(NB_DEFAULT_WINDOWS_CONFIG) protected readonly defaultWindowsConfig: NbWindowConfig,
+  ) {}
+
+  /**
+   * Opens new window.
+   * @param windowContent
+   * @param windowConfig
+   * */
+  open(
+    windowContent: TemplateRef<any> | NbComponentType,
+    windowConfig: Partial<NbWindowConfig> = {},
+  ): NbWindowRef {
+    if (this.windowsContainerViewRef == null) {
```

Comment:
` if (this.windowsContainerViewRef == null) {` =>
` if (!this.windowsContainerViewRef) {`